File: WebNovelApp/app/src/main/java/com/sinhvien/webnovelapp/models/NovelRankingDto.kt
package com.sinhvien.webnovelapp.models

import com.google.gson.annotations.SerializedName

data class NovelRankingDto(
    @SerializedName("rank") val Rank: Int,
    @SerializedName("id") val Id: Int,
    @SerializedName("title") val Title: String,
    @SerializedName("authorName") val AuthorName: String,
    @SerializedName("synopsis") val Synopsis: String,
    @SerializedName("status") val Status: String,
    @SerializedName("averageRating") val AverageRating: Double,
    @SerializedName("totalRatings") val TotalRatings: Int,
    @SerializedName("totalChapters") val TotalChapters: Int,
    @SerializedName("viewCount") val ViewCount: Long,
    @SerializedName("bookmarkCount") val BookmarkCount: Long,
    @SerializedName("wordCount") val WordCount: Long,
    @SerializedName("isPremium") val IsPremium: Boolean,
    @SerializedName("lastUpdated") val LastUpdated: String,
    @SerializedName("genres") val Genres: List<String>
)

data class NovelRankingResponse(
    @SerializedName("success") val Success: Boolean,
    @SerializedName("data") val Data: List<NovelRankingDto>,
    @SerializedName("totalCount") val TotalCount: Int = 0,
    @SerializedName("currentPage") val CurrentPage: Int = 1,
    @SerializedName("totalPages") val TotalPages: Int = 1,
    @SerializedName("pageSize") val PageSize: Int = 50,
    @SerializedName("message") val Message: String? = null
)

data class TopNovelsData(
    @SerializedName("mostViewed") val MostViewed: List<Novel>,
    @SerializedName("highestRated") val HighestRated: List<Novel>,
    @SerializedName("mostBookmarked") val MostBookmarked: List<Novel>,
    @SerializedName("mostChapters") val MostChapters: List<Novel>
)

File: DarkNovel/DarkNovel/Models/ApiModels/NovelRankingDto.cs
using System;
using System.Collections.Generic;

namespace DarkNovel.Models.ApiModels
{
    public class NovelRankingDto
    {
        public int Rank { get; set; }
        public int Id { get; set; }
        public string Title { get; set; }
        public string AuthorName { get; set; }
        public string Synopsis { get; set; }
        public string Status { get; set; }
        public decimal AverageRating { get; set; }
        public int TotalRatings { get; set; }
        public int TotalChapters { get; set; }
        public long ViewCount { get; set; }
        public long BookmarkCount { get; set; }
        public long WordCount { get; set; }
        public bool IsPremium { get; set; }
        public DateTime LastUpdated { get; set; }
        public List<string> Genres { get; set; }
    }
}